core/src/test/scala/org/dbpedia/extraction/mappings/DraftMappingExtractorTest.scala

```diff
@@ -0,0 +1,337 @@
+        ||  data9 = {{#if:{{{pushpin_map_narrow|}}}||{{#if:{{both| {{{pushpin_map|}}} | {{both|{{{latd|}}}|{{{longd|}}}}} {{both|{{{coordinates_wikidata|{{{wikidata|}}}}}}|{{#property:P625}}}} }}|""", "TestPage", lang)
+
+    val answer = Set("{{#property:P625}}")
+    var parsed = scala.collection.mutable.Set[String]()
+    for ( quad <- quads){
+      parsed.add(quad.value)
+    }
+    (parsed) should be (answer)
+  }
+
+  "DraftMappingExtractor" should """return correct property id's p742, p570, p20 """ in {
+
+    val lang = Language.English
+
+    val quads = parse(
+      """| surnom            = {{#property:p742}}
+         | date de naissance = {{Date|18|septembre|1943|au cinéma|âge=oui}}
+         | lieu de naissance = [[Montréal]], {{Québec}} ([[Canada]])
+         | nationalité       = {{drapeau|Canada}} [[Canada|Canadienne]]
+         | date de naissance = {{Date|18|septembre|1943|au cinéma|âge=oui}}
+         | lieu de naissance = [[Montréal]], {{Québec}} ([[Canada]])
+         | nationalité       = {{drapeau|Canada}} [[Canada|Canadienne]]
+         | date de décès     = {{#property:p570}}
+         | lieu de décès     = {{#property:p20}}
+         | profession        = [[Acteur|Actrice]]""", "TestPage", lang)
+
+    val answer = Set("{{#property:p742}}", "{{#property:p570}}", "{{#property:p20}}")
+    var parsed = scala.collection.mutable.Set[String]()
+    for ( quad <- quads){
+      parsed.add(quad.value)
+    }
+    (parsed) should be (answer)
+  }
+
+  "DraftMappingExtractor" should """return correct property id's P735, P734 """ in {
+
+    val lang = Language.English
+
+    val quads = parse(
+      """ | nom               = {{#property:P735}} {{#property:P734}}
+        """, "TestPage", lang)
+
+    val answer = Set("{{#property:P735}}", "{{#property:P734}}")
+    var parsed = scala.collection.mutable.Set[String]()
+    for ( quad <- quads){
+      parsed.add(quad.value)
+    }
+    (parsed) should be (answer)
+  }
+
+  "DraftMappingExtractor" should """return correct property id's p1563 """ in {
+
+    val lang = Language.English
+
+    val quads = parse(
+      """lien auteur2=Edmund Robertson
+        |jour={{{jour|}}}
+        |mois={{{mois|}}}
+        |année={{{année|}}}
+        |lire en ligne=http://www-history.mcs.st-andrews.ac.uk/Biographies/{{{id|{{#property:p1563}}}}}.html
+        |titre chapitre={{{title|{{PAGENAME}}}}}
+        |titre ouvrage=[[MacTutor History of Mathematics archive]]
+        |lien éditeur=université de St Andrews
+        |éditeur=université de {{lang|en|St Andrews}}
+        |consulté le={{{accessdate|}}}""", "TestPage", lang)
+
+    val answer = Set("{{#property:p1563}}")
+    var parsed = scala.collection.mutable.Set[String]()
+    for ( quad <- quads){
+      parsed.add(quad.value)
+    }
+    (parsed) should be (answer)
+  }
+
+  "DraftMappingExtractor" should """return correct property id's P31, P570 """ in {
+
+    val lang = Language.English
+
+    val quads = parse(
```

Comment:
It will make testing easier if at this stage you test only a function that takes wikitext as an argument and returns a set of (TermplateName, TemlateParameter, WikidataPropertyID).

so for now you can forget the complete extractor, quads etc and focus only on this specific input / output